lib/ansible/modules/cloud/amazon/ec2_vpc_endpoint.py

```diff
@@ -366,16 +359,14 @@ def main():
                 module.fail_json(msg="Error - no region provided")
         else:
             module.fail_json(msg="Can't retrieve connection information - " + str(e),
-                             exception=traceback.format_exc(),
-                             **camel_dict_to_snake_dict(e.response))
+                             exception=traceback.format_exc())
 
     try:
         region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
         ec2 = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_kwargs)
     except botocore.exceptions.NoCredentialsError as e:
         module.fail_json(msg="Failed to connect to AWS due to wrong or missing credentials: %s" % str(e),
-                         exception=traceback.format_exc(),
-                         **camel_dict_to_snake_dict(e.response))
+                         exception=traceback.format_exc())
```

Comment:
I added 4d4553bbb035ee1d49bff7925876c1dd88282bb6 so you can remove the try/except and just keep the contents of the try.